apps/prairielearn/src/models/course.ts

```diff
@@ -12,3 +24,50 @@ export async function selectCourseById(course_id: string): Promise<Course> {
     CourseSchema,
   );
 }
+
+export async function selectAuthorizedCourses({
+  user_id,
+  is_administrator,
+  current_course,
+  authz_data_overrides,
+}: {
+  user_id: string;
+  is_administrator: boolean;
+  current_course: Course;
+  authz_data_overrides: any | null | undefined;
+}) {
+  const { courses } = await callValidatedOneRow(
+    'courses_with_staff_access',
+    [user_id, is_administrator],
+    z.object({
+      courses: z.array(CourseWithPermissionsSchema),
+    }),
+  );
+
+  // If any overrides are in place, we'll confine the user to the current course.
+  if (authz_data_overrides) {
+    return courses.filter((c) => idsEqual(c.id, current_course.id));
+  }
```

Comment:
This could use a close look to make sure I've correctly translated the logic from `authzCourseOrInstance`.